src/entity/active_model.rs

```diff
@@ -65,9 +67,71 @@ pub trait ActiveModelTrait: Clone + Debug {
 
     fn default() -> Self;
 
-    // below is not yet possible. right now we define these methods in DeriveActiveModel
-    // fn save(self, db: &DatabaseConnection) -> impl Future<Output = Result<Self, DbErr>>;
-    // fn delete(self, db: &DatabaseConnection) -> impl Future<Output = Result<DeleteResult, DbErr>>;
+    async fn insert(self, db: &DatabaseConnection) -> Result<Self, DbErr>
+    where
+        <Self::Entity as EntityTrait>::Model: IntoActiveModel<Self>,
+    {
+        let am = self;
+        let exec = <Self::Entity as EntityTrait>::insert(am).exec(db);
+        let res = exec.await?;
+        // TODO: if the entity does not have auto increment primary key, then last_insert_id is a wrong value
+        // FIXME: Assumed valid last_insert_id is not equals to Default::default()
+        if <<Self::Entity as EntityTrait>::PrimaryKey as PrimaryKeyTrait>::auto_increment()
+            && res.last_insert_id != <<Self::Entity as EntityTrait>::PrimaryKey as PrimaryKeyTrait>::ValueType::default()
+        {
+            let find = <Self::Entity as EntityTrait>::find_by_id(res.last_insert_id).one(db);
+            let found = find.await;
+            let model: Option<<Self::Entity as EntityTrait>::Model> = found?;
```

Comment:
We need to add `ModelTrait: Sync` because of this

Reply:
🤔 `Send` is fair I guess. `Sync` would be too harsh.

Reply:
`Send` I mean loll It's a typo